models/base.rb

```diff
@@ -60,6 +60,17 @@ def vrp_result_fields
       end
     end
 
+    def duplicate_safe(options = {})
+      object_hash = JSON.parse(self.to_json, symbolize_names: true)
+
+      options.each{ |key, value|
+        next unless object_hash.key?(key)
+
+        object_hash[key] = value
```

Comment:
If we decide to not to dump the keys with default values this will lead to an issue, we need to check if `key` exists as a filed or a member function via the original object `self` itself.